pkg/image/api/v1/types.go

```diff
@@ -42,6 +44,73 @@ type ImageLayer struct {
+	Type string `json:"type"`
+	// An opaque binary string which is an image's signature.
+	Content []byte `json:"content"`
+	// Conditions represent the latest available observations of a signature's current state.
+	Conditions []SignatureCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
+
+	// Following metadata fields will be set by server if the signature content is successfully parsed and
+	// the information available.
+
+	// A human readable string representing image's identity. It could be a product name and version, or a
+	// image pull spec.
+	ImageIdentity string `json:"imageIdentity,omitempty"`
+	// Contains claims from the signature.
+	SignedClaims map[string]string `json:"signedClaims,omitempty"`
+	// If specified, it is the time of signature's creation.
+	Created *unversioned.Time `json:"created,omitempty"`
+	// If specified, it holds information about an issuer of signing certificate or key (a person or entity
+	// who signed the signing certificate or key).
+	IssuedBy SignatureIssuer `json:"issuedBy,omitempty"`
+	// If specified, it holds information about a subject of signing certificate or key (a person or entity
+	// who signed the image).
+	IssuedTo SignatureSubject `json:"issuedTo,omitempty"`
+}
+
+/// SignatureConditionType is a type of image signature condition.
+type SignatureConditionType string
+
+// SignatureCondition describes an image signature condition of particular kind at particular probe time.
+type SignatureCondition struct {
+	// Type of job condition, Complete or Failed.
+	Type SignatureConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status kapi.ConditionStatus `json:"status"`
+	// Last time the condition was checked.
+	LastProbeTime unversioned.Time `json:"lastProbeTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime unversioned.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+// SignatureGenericEntity holds a generic information about a person or entity who is an issuer or a subject
+// of signing certificate or key.
+type SignatureGenericEntity struct {
+	// Organization name.
+	Organization string `json:"organization,omitempty"`
+	// Common name (e.g. openshift-signing-service).
+	CommonName string `json:"commonName,omitempty"`
+}
+
+// SignatureIssuer holds information about an issuer of signing certifikate or key.
+type SignatureIssuer struct {
+	SignatureGenericEntity `json:",inline"`
+}
+
+// SignatureSubject holds information about a person or entity who created the signature.
+type SignatureSubject struct {
+	SignatureGenericEntity `json:",inline"`
+	// If present, it is a fingerprint of public key belonging to the subject used to verify image signature.
```

Comment:
> Add an example of a fingerprint, and any other detail that is required to understand why it is used.

Added an example.

> Why is this not PublicKeyFingerprint if it's a fingerprint?

I corrected the godoc. Since this serves solely an informative purpose, I find whole fingerprint too long to display. The long key ID (64bits) prefixed with `0x` should be appropriate for this purpose. (e.g. `0x685ebe62bf278440`).